plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/interactive/InteractiveUtil.xtend

```diff
@@ -0,0 +1,168 @@
+     * is different to the value on the ElkNode.
+     * If the new value on the KNode was the default value of the property 
+     * then the property is set to null on the ElkNode.
+     * @param elkNode The target ElkNode
+     * @param kNode The source KNode of the property
+     * @param prop Determines which IProperty should be copied
+     */
+    static def <T> copyConstraintProp(ElkNode elkNode, KNode kNode, IProperty<T> prop) {
+        val kNodeValue = kNode.getProperty(prop)
+
+        if (elkNode.getProperty(prop) !== kNodeValue) {
+            if (kNodeValue === prop.^default) {
+                elkNode.setProperty(prop, null)
+            } else {
+                elkNode.setProperty(prop, kNodeValue)
+
+            }
+        }
+
+    }
+
+    /**
+     * Copies constraint properties depending on the algorithm from kNode to elkNode 
+     * by using {@code copyConstraintProp()}.
+     * 
+     * @param elkNode The target ElkNode
+     * @param kNode The source KNode of the property
+     */
+    static def copyAllConstraints(ElkNode elkNode, KNode kNode) {
+        val algorithm = kNode.parent.getProperty(CoreOptions.ALGORITHM)
+        var List<IProperty<?>> props = #[]
+        if(algorithm == null || algorithm == 'layered') {
+            props = #[
+                LayeredOptions.LAYERING_LAYER_CHOICE_CONSTRAINT,
+                LayeredOptions.CROSSING_MINIMIZATION_POSITION_CHOICE_CONSTRAINT
+            ]
+        } else if ("rectpacking".equals(algorithm)) {
+            props = #[
+                RectPackingOptions.DESIRED_POSITION,
+                RectPackingOptions.ASPECT_RATIO
+            ]
+        }
+        for (prop : props) {
+            copyConstraintProp(elkNode, kNode, prop)
+        }
+
+    }
+
+    /**
+     * Determines the root of the given node.
+     * 
+     * @param node ElkNode, which root should be returned.
+     */
+    static def getRootNodeOf(ElkNode node) {
+        var ElkNode parent = node
+        while (parent.parent !== null) {
+            parent = parent.parent
+        }
+        return parent
+    }
+
+    /**
+     * Determines the root of the given node.
+     * 
+     * @param node KNode, which root should be returned.
+     */
+    static def getRootNodeOf(KNode node) {
+        var KNode parent = node
+        while (parent.parent !== null) {
+            parent = parent.parent
+        }
+        return parent
+    }
+
+    /**
+     * Collects the adjacent nodes of {@code node} in a list.
+     * @param node The node of which you want to know the adjacent nodes.
+     * @return 
+     */
+    static def getAdjacentNodes(KNode node) {
+        val inEdges = node.incomingEdges
+        val outEdges = node.outgoingEdges
+        var List<KNode> adjacentNodes = newArrayList()
+
+        for (e : inEdges) {
+            adjacentNodes.add(e.source)
+        }
+        for (e : outEdges) {
+            adjacentNodes.add(e.target)
+        }
+        return adjacentNodes
+    }
+}
```

Comment:
This as well.

Reply:
Do you mean it is there or do you suggest to add it there?